### PR TITLE
Create sending-and-receiving-messages

### PR DESCRIPTION
**Pull request by JorgePe**
This is a first draft of a tutorial on how to use MQTT protocol to exchange messages with ev3dev devices.
It totally lacks formatting, need to learn how to do that, but I think is is usable.
I also intend to complete it with a better, more complex, example.

### CODE REVIEW

**Comment by WasabiFan** [comment]
I get the feeling that this fell through the cracks and has been forgotten about (I definitely forgot about it). @dlech?

File: docs/tutorials/sending-and-receiving-messages [hunk 1]
```diff
@@ -0,0 +1,174 @@
+---
+title: Sending and receiving messages with MQTT
```
Comment (dlech) [inline docs/tutorials/sending-and-receiving-messages]: Title case please:

    Sending and Receiving Messages with MQTT

File: docs/tutorials/sending-and-receiving-messages [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+subject: Networking
+---
+
+* Table of Contents
+{:toc}
+
+## Intro
+
+«MQTT (Message Queue Telemetry Transport) is an ISO standard (ISO/IEC PRF 20922)
```
Comment (dlech) [inline docs/tutorials/sending-and-receiving-messages]: « is not needed

also a link to wikipedia or something for more info about MQTT would be nice.

File: docs/tutorials/sending-and-receiving-messages [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+publish-subscribe based "light weight" messaging protocol for use on top of the
+TCP/IP protocol. It is designed for connections with remote locations where a
+"small code footprint" is required or the network bandwidth is limited.»
```
Comment (dlech) [inline docs/tutorials/sending-and-receiving-messages]: same with »

File: docs/tutorials/sending-and-receiving-messages [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+from anything that can publish MQTT messages like a PC, a mobile phone, a
+Raspberry Pi or an Arduino. Welcome to the Internet of Things!
+
+One interesting possibility is running `node-red` on our PC as it offers two
+built-in MQTT nodes ("mqtt in" and "mqtt out"). With just a few clicks we can
+create a flow that connects our EV3's to the internet. Of course, we can also run
+`node-red` on our EV3 but will probably exceed our little fellow resources (but
+a Raspberry Pi and a BrickPi can run `node-red` and `mosquitto` easily).
+
+A more advanced example is being prepared.
```
Comment (dlech) [inline docs/tutorials/sending-and-receiving-messages]: Probably best to leave out "A more advanced example is being prepared." Just add it when it is ready.

File: docs/tutorials/sending-and-receiving-messages [hunk 5]
```diff
@@ -0,0 +1,174 @@
+---
+title: Sending and receiving messages with MQTT
+subject: Networking
+---
```
Comment (dlech) [inline docs/tutorials/sending-and-receiving-messages]: you can add yourself as an author if you like

    author: "@JorgePe"

**Comment by dlech** [comment]
This looks really great. If you want someone to have a look after you add more commits, you need to leave a comment. GitHub does not send out notifications when commits are pushed.

Also, please change the file name to match the title: `sending-and-receiving-messages-with-mqtt`.

P.S. Saw the video. :wink:

**Comment by dlech** [comment]
>How do I add a jpeg file? Not the link, the file itself?

I don't think you can do this through the web browser. You have to use command line git on your computer. They go somewhere in <https://github.com/ev3dev/ev3dev.github.io/tree/master/images>.

File: docs/tutorials/sending-and-receiving-messages-with-mqtt [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+* Table of Contents
+{:toc}
+
+## Intro
+
+[MQTT](https://en.wikipedia.org/wiki/MQTT) (Message Queue Telemetry Transport) is an ISO standard (ISO/IEC PRF 20922)
+publish-subscribe based "light weight" messaging protocol for use on top of the
+TCP/IP protocol. It is designed for connections with remote locations where a
+"small code footprint" is required or the network bandwidth is limited.»
```
Comment (dlech) [inline docs/tutorials/sending-and-receiving-messages-with-mqtt]: Missed removing the » here

**Comment by dlech** [comment]
then to add the image to the page, we usually use this:

    {% include screenshot.html source="/images/Website/nfs_on_ev3dev.png" %}

**Comment by JorgePe** [comment]
OK, I'll try to add one or two photos/screenshots later on - this week it's more easy for me to add just a link to youtube video.

**Comment by dlech** [comment]
http://www.ev3dev.org/docs/tutorials/sending-and-receiving-messages-with-mqtt/

FYI, there are a couple things I didn't noticed until after I merged that I fixed. The file extension needs to be `.md`. I also added bullets to the `Basics` section.

**Comment by JorgePe** [comment]
Thanks!
Didn't hurt much after all :) Will try to contribute with more tutorials or even a project.

**Pull request merged**